components/content-service-api/initializer.proto

```diff
@@ -115,10 +115,11 @@ message SnapshotInitializer {
 }
 
 // A prebuild initializer combines snapshots with Git: first we try the snapshot, then apply the Git clone target.
-// If restoring the snapshot fails, we fall back to a regular Git initializer.
+// If restoring the snapshot fails, we fall back to a regular Git initializer, which might be composite git initializer for multi-repo projects.
 message PrebuildInitializer {
     SnapshotInitializer prebuild = 1;
     GitInitializer git = 2;
+    CompositeInitializer composite = 3;
```

Comment:
```suggestion
    CompositeInitializer sub_repos = 3;
```

Reply:
The composite initializer doesn not only contain the subrepos, but all of them.
We can name the property `multi_git` if that helps your concern?

Reply:
Wouldn't the `git` field `2` contain the initializer of the parent repo, making the rest just sub_repos?

Reply:
If that's not the case, then `multi_git` makes sense

Reply:
> Wouldn't the git field 2 contain the initializer of the parent repo, making the rest just sub_repos?

No, if it's a multi-repo a composite initialize is used otherwise a single GitInitializer

Reply:
In that case from an API perspective I'd have expected the message to look like:
```protobuf
message PrebuildInitializer {
    SnapshotInitializer prebuild = 1;
    oneof git {
        GitInitializer single = 2;
        CompositeInitializer composite = 3;
    }
}
```

AFAIK that would even be backwards compatible: https://developers.google.com/protocol-buffers/docs/proto3#tag_reuse_issues